ecs_client_test.go

```diff
@@ -39,49 +60,205 @@ func TestECSClientRegisterAndDeregisterTaskDefinition(t *testing.T) {
 		HTTPClient: hc,
 	})
 	require.NoError(t, err)
+	require.NotNil(t, c)
 
-	t.Run("RegisterFailsWithInvalidInput", func(t *testing.T) {
-		out, err := c.RegisterTaskDefinition(ctx, &ecs.RegisterTaskDefinitionInput{})
-		assert.Error(t, err)
-		assert.Zero(t, out)
-	})
-	t.Run("DeregisterFailsWithInvalidInput", func(t *testing.T) {
-		out, err := c.DeregisterTaskDefinition(ctx, &ecs.DeregisterTaskDefinitionInput{})
-		assert.Error(t, err)
-		assert.Zero(t, out)
+	registerOut, err := c.RegisterTaskDefinition(ctx, &ecs.RegisterTaskDefinitionInput{
+		ContainerDefinitions: []*ecs.ContainerDefinition{
+			{
+				Command: []*string{aws.String("echo"), aws.String("foo")},
+				Image:   aws.String("ubuntu"),
+				Name:    aws.String("print_foo"),
+			},
+		},
+		Cpu:    aws.String("128"),
+		Memory: aws.String("4"),
+		Family: aws.String(os.Getenv("AWS_ECS_TASK_DEFINITION_PREFIX") + "bar"),
```

Comment:
Use a helper to include the task definition prefix and randomize the task definition name like in [`makeTestSecret`](https://github.com/evergreen-ci/cocoa/blob/3ef0df31695badfaada55fb893ae5d019f79961d/secret/secrets_manager_client_test.go#L190-L192).